tests/41end-to-end-keys/06-device-lists.pl

```diff
@@ -233,6 +233,163 @@ sub sync_until_user_in_device_list
       });
    };
 
+test "See devices for invited users via /keys/query",
+   requires => [ local_user_fixture(), remote_user_fixture(),
+                 qw( can_upload_e2e_keys )],
+
+   check => sub {
+      my ( $user1, $user2 ) = @_;
+
+      my $room_id;
+
+      matrix_create_room( $user1 )->then( sub {
+         ( $room_id ) = @_;
+         matrix_put_e2e_keys( $user2 );
+      })->then( sub {
+         matrix_sync( $user1 );
+      })->then( sub {
+         matrix_invite_user_to_room( $user1, $user2, $room_id )
+      # })->then( sub {
+      #    # KNOWN BROKEN: (https://github.com/matrix-org/synapse/issues/3503)
```

Comment:
I don't really understand what you're doing here. If you just comment it out and run the query anyway, surely it is going to fail, because the update might not yet have propagated over federation?